src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -59,12 +59,14 @@
    metrics-service :- (schema/protocol jruby-metrics/JRubyMetricsService)
    max-queued-requests :- schema/Int
    max-retry-delay :- schema/Int]
-  (fn [request]
-    (let [metrics-map (jruby-metrics/get-metrics metrics-service)
-          requested-count (-> metrics-map :requested-instances deref count)]
-      (if (>= requested-count max-queued-requests)
-        (let [err-msg (i18n/trs "The number of requests waiting for a JRuby instance has exceeded the limit allowed by the max-queued-requests setting.")
-              response (output-error request {:msg err-msg} 503)]
+  (let [metrics-map (jruby-metrics/get-metrics metrics-service)
```

Comment:
Are the elements in the metrics-map mutable? If not, lifting them out of the function seems like they will be set only on startup and then never updated again.

Reply:
Looks like queue-limit-hit-meter is a metrics object and requested-instances is an atom.

Reply:
@Sharpie This looks like a safe change, but was there a reason for it?

Reply:
ah never mind. i can read commit messages i swear.

Reply:
The metrics map comes from the service context of the JRubyMetricsService. Should be immutable with the possible exception of a new state generated during reloads. The original commit that added rate limiting, PR #1365, also added a dependency on JRubyMetricsService to the RequestHandlerService which calls this function during init to construct a ring handler. So, both the atom and meter should be immutable within the closure of the ring handler.